cpp/tests/utilities/identify_stream_usage.cpp

```diff
@@ -66,6 +67,34 @@ rmm::cuda_stream_view const get_default_stream()
 }  // namespace test
 #endif
 
+#ifdef STREAM_MODE_TESTING
+namespace detail {
+
+/**
+ * @brief Implementation of `cuda_stream_pool` that always returns the
+ * `cudf::test::get_default_stream()`
+ */
+class test_cuda_stream_pool : public cuda_stream_pool {
+ public:
+  rmm::cuda_stream_view get_stream() override { return cudf::test::get_default_stream(); }
+  rmm::cuda_stream_view get_stream(stream_id_type stream_id) override
```

Comment:
Can we remove the variable name since it's never used (or if that causes unforeseen problems, mark it as `[[ maybe_unused ]]`)?
```suggestion
  rmm::cuda_stream_view get_stream(stream_id_type) override
```